Backend/lib/message-service/src/Lib/Domain/Action/UI/Message.hs

```diff
@@ -114,22 +112,22 @@ getMessage (driverId, _) messageId = do
             mediaFiles = mediaFilesApiType
           }
 
-fetchMedia :: (Id SP.Person, Id DM.Merchant) -> Text -> Flow Text
-fetchMedia (driverId, _) filePath = do
-  _ <- Esq.runInReplica $ QP.findById driverId >>= fromMaybeM (PersonNotFound driverId.getId)
+fetchMedia :: (MonadFlow m, MonadReader r m, HasField "s3Env" r (S3.S3Env m)) => (Id Domain.Person, Id Domain.Merchant) -> Text -> m (Maybe a) -> m Text
+fetchMedia (driverId, _) filePath findByPersonId = do
+  _ <- findByPersonId >>= fromMaybeM (PersonNotFound driverId.getId)
   S3.get $ T.unpack filePath
 
-messageSeen :: (Id SP.Person, Id DM.Merchant) -> Id Domain.Message -> Flow APISuccess
+messageSeen :: (MonadFlow m, MonadReader r m, EsqDBReplicaFlow m r, HasField "esqDBEnv" r EsqDBEnv, HasField "loggerEnv" r LoggerEnv) => (Id Domain.Person, Id Domain.Merchant) -> Id Domain.Message -> m APISuccess
 messageSeen (driverId, _) messageId = do
   messageDetails <- Esq.runInReplica $ MRQ.findByMessageIdAndDriverId messageId (cast driverId) >>= fromMaybeM (InvalidRequest "Message not found")
   Esq.runTransaction $ do
     when (not messageDetails.readStatus) $ MQ.updateMessageViewCount messageId 1
     MRQ.updateSeenAndReplyByMessageIdAndDriverId messageId (cast driverId) True Nothing
   return Success
 
-messageLiked :: (Id SP.Person, Id DM.Merchant) -> Id Domain.Message -> Flow APISuccess
-messageLiked (driverId, _) messageId = do
-  _ <- Esq.runInReplica $ QP.findById driverId >>= fromMaybeM (PersonNotFound driverId.getId)
+messageLiked :: (HasField "language" a (Maybe Language), MonadFlow m, EsqDBReplicaFlow m r, EsqDBFlow m r) => (Id Domain.Person, Id Domain.Merchant) -> Id Domain.Message -> m (Maybe a) -> m APISuccess
+messageLiked (driverId, _) messageId findByPersonId = do
+  _ <- findByPersonId >>= fromMaybeM (PersonNotFound driverId.getId)
```

Comment:
I'm pretty sure that these checks should be part of main app and not a part of lib

Reply:
I can't use same Flow here as it wrapper function return type is Flow, so here have to make the return type monadic hence these checks are required.

Reply:
We should pass the person details here, instead of passing function and calling here?